src/pathfinding_service/model/token_network.py

```diff
@@ -76,19 +111,36 @@ def __init__(
         self.value = value
         self.address_to_reachability = address_to_reachability
         self.fees: List[FeeAmount] = []
-        self._calc_fees()
+        self._calculate_fees()
 
-        log.debug("Creating Path object", nodes=nodes, is_valid=self.is_valid)
+        log.debug("Creating Path object", nodes=nodes, is_valid=self.is_valid, fees=self.fees)
 
-    def _calc_fees(self) -> None:
-        total = self.value
+    def _calculate_fees(self) -> None:
+        total = PaymentWithFeeAmount(self.value)
         try:
             for prev_node, mediator, next_node in reversed(list(window(self.nodes, 3))):
-                fee_for_this_mediator = self.G[prev_node][mediator]["view"].forward_fee_receiver(
-                    total
-                ) + self.G[mediator][next_node]["view"].forward_fee_sender(total)
-                total += fee_for_this_mediator
-                self.fees.append(fee_for_this_mediator)
+                view_in: ChannelView = self.G[prev_node][mediator]["view"]
+                view_out: ChannelView = self.G[mediator][next_node]["view"]
+
+                fee_out = view_out.backwards_fee_sender(
+                    balance=Balance(view_out.capacity), amount=total
+                )
+                if fee_out is None:
```

Comment:
Depending on whether (and how) we share this code with the client, we could just let the `UndefinedMediationFee` exception bubble up and skip the special case, here.

Reply:
Good point, as it looks now we could remove all `try except` blocks in the functions and catch it toplevel. But let's discuss tomorrow how we can reuse the client code.

Reply:
Let's keep this code short term to focus on the scenarios. Medium term we want to re-use the code from the client, so that there's only one implementation.